remove finding of pods by IP

sschepens opened the pull request:
**Please provide a description of this PR:**
Creating this as a draft to check if tests pass, I'm not sure why we still maintain this logic, I cannot understand in which cases we would fail to find a Pod by it's name+namespace but find them by IP.

sschepens commented:
/test all

istio-testing commented:
@sschepens: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
release-notes_istio | 3d592477a2a54744a9b88585db4a44ddd70a15bd | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/56502/release-notes_istio/1930228328959053824) | true | `/test release-notes`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@sschepens: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
release-notes_istio | 3d592477a2a54744a9b88585db4a44ddd70a15bd | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/56502/release-notes_istio/1930228328959053824) | true | `/test release-notes`
unit-tests-arm64_istio | 3d592477a2a54744a9b88585db4a44ddd70a15bd | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/56502/unit-tests-arm64_istio/1930228330636775424) | true | `/test unit-tests-arm64`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@sschepens: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
release-notes_istio | 3d592477a2a54744a9b88585db4a44ddd70a15bd | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/56502/release-notes_istio/1930228328959053824) | true | `/test release-notes`
unit-tests-arm64_istio | 3d592477a2a54744a9b88585db4a44ddd70a15bd | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/56502/unit-tests-arm64_istio/1930228330636775424) | true | `/test unit-tests-arm64`
unit-tests_istio | 3d592477a2a54744a9b88585db4a44ddd70a15bd | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/56502/unit-tests_istio/1930228331471441920) | true | `/test unit-tests`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

sschepens commented:
/test all

istio-testing commented:
@sschepens: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
unit-tests-arm64_istio | 3d592477a2a54744a9b88585db4a44ddd70a15bd | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/56502/unit-tests-arm64_istio/1930228330636775424) | true | `/test unit-tests-arm64`
unit-tests_istio | 3d592477a2a54744a9b88585db4a44ddd70a15bd | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/56502/unit-tests_istio/1930228331471441920) | true | `/test unit-tests`
integ-ambient-mc_istio | 3d592477a2a54744a9b88585db4a44ddd70a15bd | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/56502/integ-ambient-mc_istio/1930228317382774784) | false | `/test integ-ambient-mc`
release-notes_istio | 72ec1b962331b91d122700f353b5ff50c13e63d5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/56502/release-notes_istio/1930246102410334208) | true | `/test release-notes`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ramaraochavali commented:
https://github.com/istio/istio/pull/28992/files#r535394193 - reason why PodbyIP is still used

sschepens commented:
> > Don't know how many peope customize proxies to the point that the proxy ID is not in the form we would expect.
> 
> 
> 
> It is quite possible. I have seen folks do that with a non injected proxy with a custom bootstrap pointing to Istiod for controlplane information.
> 
> 
> 
> 


We could add an Upgrade Note for next version warning of this change and make them send the correct metadata

istio-testing commented:
@sschepens: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
integ-ambient-mc_istio | 72ec1b962331b91d122700f353b5ff50c13e63d5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/56502/integ-ambient-mc_istio/1930246090330738688) | false | `/test integ-ambient-mc`
release-notes_istio | 72ec1b962331b91d122700f353b5ff50c13e63d5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/56502/release-notes_istio/1930323837673541632) | true | `/test release-notes`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

sschepens commented:
Added a feature flag to restore the origina behaviour but it's turned off by default, also added an Upgrade Note to warn users, do you think this is enough?

istio-testing commented:
@sschepens: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
integ-ambient-mc_istio | b8aaffd6a565ac0bf2ed8bbb4e396f43e293d80d | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/56502/integ-ambient-mc_istio/1930606898327851008) | false | `/test integ-ambient-mc`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ramaraochavali reviewed (commented):
Thanks. Curios, Did you run in to any issue and that is wanted to fix or as a general cleanup?

sschepens commented:
We found an issue in which apparently our informers were delayed and a new pod came up using the same IP address as a previous one, this caused this piece of code to not find the pod by name and search it by IP, returning an incorrect pod. The most basic fix is https://github.com/istio/istio/pull/56501 but it's technically not even correct to do that, even if it's in the same namespace it's probably not the pod we want.

In our case this ended up making Istiod take the Labels from an incorrect pod and ended up pushing configurations that shouldn't have been pushed to that proxy.

ramaraochavali commented:
ok. Thanks. @hzxuzhonghu FYI
